M365/MDO/EmailAuthChecker.ps1

```diff
@@ -0,0 +1,7180 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# email providers
+#cspell:words gmail, googlemail, amazonses, proofpoint, pphosted, mimecast
+
+# DKIM Providers
+#cspell:words mailchimp, smtpapi, mailgun, mailo, scph, zendesk, salesforce, klaviyo, aweber, getresponse, convertkit, infusionsoft, infusionsoft, pardot, marketo, eloqua, eloqua, sendlane, moosend, omnisend, emailoctopus, sendinblue, elasticemail, pepipost, socketlabs, mailjet, dynadot, zoho, protonmail, fastmail, rackspace, bluehost, namecheap, plesk
+
+# protocol words
+#cspell:words softfail, softpass, permerror, temperror, compauth, adkim, aspf, NSPM, BIMP, DIMP, FTBP, HPHSH, HPHISH, HSPM, INTOS, MALW, OSPM, PHSH, SPOOF, UIMP, dmarc, domainkey, mxvault
+
+#html tags
+#cspell:words onclick, thead, tbody, colgroup, mouseleave, mouseenter, ctry, darr, minmax, rgba, nowrap, uarr, onmouseover, onmouseout, linecap, dashoffset, dasharray
```

Comment:
If the word is not case sensitive, and can use camel casing or pascal casing, then use that. 

If it is Case Sensitive, you can use ToLower() when required as well for actual words that aren't going to trigger an issue with our spell checker.